web-console/src/dialogs/doctor-dialog/doctor-checks.tsx

```diff
@@ -0,0 +1,421 @@
+];
+
+export const DOCTOR_CHECKS: DoctorCheck[] = [
+  // -------------------------------------
+  // Self (router) checks
+  // -------------------------------------
+  {
+    name: 'Verify own status',
+    check: async controls => {
+      // Make sure that the router responds to /status and gives some valid info back
+      let status: any;
+      try {
+        status = (await axios.get(`/status`)).data;
+      } catch (e) {
+        controls.addIssue(
+          `Did not get a /status response, is the cluster running? Got: ${e.message}`,
+        );
+        controls.terminateChecks();
+        return;
+      }
+
+      if (typeof status.version !== 'string') {
+        controls.addIssue('Could not get a valid /status response.');
+      }
+    },
+  },
+  {
+    name: 'Verify own runtime properties',
+    check: async controls => {
+      // Make sure that everything in /status/properties is above board
+      let properties: Record<string, string>;
+      try {
+        properties = (await axios.get(`/status/properties`)).data;
+      } catch (e) {
+        controls.addIssue('Did not get a /status/properties response, something must be broken.');
+        return;
+      }
+
+      // Check that the management proxy is on, it really should be for someone to access the console in the first place but everything could happen
+      if (properties['druid.router.managementProxy.enabled'] !== 'true') {
+        controls.addIssue(
+          `The router's "druid.router.managementProxy.enabled" is not reported as "true" that is unusual.`,
+        );
+      }
+
+      // Check that the underlying Java is Java 8 the only officially supported Java version at the moment.
+      if (
+        properties['java.runtime.version'] &&
+        !properties['java.runtime.version'].startsWith('1.8')
+      ) {
+        controls.addSuggestion(
+          `It looks like are running Java ${properties['java.runtime.version']}, Druid only officially supports Java 1.8.x`,
+        );
+      }
+
+      // Check that "user.timezone"
+      if (properties['user.timezone'] && properties['user.timezone'] !== 'UTC') {
+        controls.addSuggestion(
+          `It looks like "user.timezone" is set to ${properties['user.timezone']}, it is recommended to set this to "UTC"`,
+        );
+      }
+    },
+  },
+
+  // -------------------------------------
+  // Coordinator and Overlord
+  // -------------------------------------
+  {
+    name: 'Verify the Coordinator and Overlord status',
+    check: async controls => {
+      // Make sure that everything in Coordinator's /status is good
+      let myStatus: any;
+      try {
+        myStatus = (await axios.get(`/status`)).data;
+      } catch {
+        return;
+      }
+
+      let coordinatorStatus: any;
+      try {
+        coordinatorStatus = (await axios.get(`/proxy/coordinator/status`)).data;
+      } catch (e) {
+        controls.addIssue('Did not get a /status response from the coordinator, is it running?');
```

Comment:
Hmm I think it'd be nice to be consistent about the format of the checks. I'd suggest something like: Potential issue. Optional clarification of why it may or may not be an issue. Suggestion, if any.

In this case that'd be:

> Did not get a /status response from the Coordinator. Try confirming that it is running and accessible.

Btw, these error message should be variable based on the specific error. 404 is different from 400/401/500 etc.

Btw also, please be consistent about capitalization (IMO I like capitalizing the process types).